lib/player.js

```diff
@@ -685,6 +703,65 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     if (mediaElement) {
       this.attach(mediaElement, /* initializeMediaSource= */ true);
     }
+
+    if (canvas) {
+      this.lcevcCanvas_ = canvas;
+    }
+  }
+
+  /**
+   * Create a shaka.lcevc.Dil object
+   * @private
+   */
+  createDIL_() {
+    if (this.lcevcDil_ == null) {
+      this.lcevcDil_ = new shaka.lcevc.Dil(
+          /** @type {HTMLVideoElement} */ (this.video_),
+          this.lcevcCanvas_,
+          this.lcevcDilConfig_,
+      );
+      if (this.mediaSourceEngine_) {
+        this.mediaSourceEngine_.updateLcevcDil(this.lcevcDil_);
+      }
+    }
+  }
+
+  /**
+   * Close a shaka.lcevc.Dil object if present and hide the canvas.
+   * @private
+   */
+  closeDIL_() {
+    if (this.lcevcDil_ != null) {
+      this.lcevcDil_.release();
+      this.lcevcDil_ = null;
+    }
+  }
+
+  /**
+   * Setup shaka.lcevc.Dil object
+   * @private
+   */
+  setupLcevc_(config) {
+    const safariVersion = shaka.util.Platform.safariVersion();
+    if (safariVersion) {
+      if (config.streaming.useNativeHlsOnSafari) {
+        shaka.log.alwaysWarn('LCEVC Error: Only Safari with MSE enabled ' +
```

Comment:
this.setupLcevc_() is now added to onSrcEquals_ too.

Reply:
You can't use LCEVC with src=.  Without MediaSource playback, we can't send the necessary information into the DIL.

So you don't really even need this check here.  If you want to warn about LCEVC being non-functional with src=, do it inside the src= setup instead of here.  There's no point in calling this from src= or checking for Safari version anywhere.  You can use src= mode with plain mp4s on any browser, not just for Safari and native HLS.